src/proof_system/widget/mod.rs

```diff
@@ -189,6 +198,7 @@ impl VerifierKey {
     }
 }
 
+#[cfg(feature = "alloc")]
```

Comment:
The whole struct `ProverKey` with the implementation looks like `alloc` only so it might be under a sub module as mentioned in previous comments.